enhancements/cluster-logging/loki-tokenized-auth-enablement.md

```diff
@@ -0,0 +1,343 @@
+3. console prompts user to create roles+supply ARN-type string
+4. console creates subscription with ARN-type string embedded as a spec.config.env
+5. operator deployment is created with ARN-type string in env
+6. operator creates `CredentialsRequest` including the ARN-type string
+7. Cloud Credential Operator populates Secret based on `CredentialsRequest`
+8. operator loads Secret, makes cloud service requests
+
+For the LokiStack admnistator using the OpenShift CLI - Operator installation under the proposed system:
+
+- For AWS:
+```yaml
+kind: Subscription
+metadata:
+ name: ...
+spec:
+  config:
+    env:
+    - name: ROLEARN
+      value: "<role ARN >"
+```
+
+- For Azure:
+```yaml
+kind: Subscription
+metadata:
+ name: ...
+spec:
+  config:
+    env:
+    - name: CLIENT_ID
+      value: "<azure client id>"
+    - name: TENANT_ID
+      value: "<azure tenant id>"
+    - name: SUBSCRIPTION_ID
+      value: "<azure subscription id>""
+```
+
+- For GCP: TBD along 4.15.
+
+For the Loki Operator author team:
+- Add code to create a `CredentialRequest` per `LokiStack` resource.
+- Add IAM-idetifier-type fields to the `CredentialRequest` resources.
+- Add eventing to report status on a CR to indicate lacking STS credentials for fully operational deploy or update.
+
+#### AWS STS Support
+
+The Loki Operator will use the `ROLEARN` environment variable and create a `CredentialsRequest` resource in the CCO operator namespace per LokiStack resource. The `ROLEARN` value is placed in the `spec.providerSpec.stsIAMRoleARN` field. The `CredentialsRequest` resource references the main LokiStack serviceaccount name that is the same as the LokiStack resource name.
+
+```yaml
+apiVersion: cloudcredential.openshift.io/v1
+kind: CredentialsRequest
+metadata:
+  name: logging-loki-aws-creds
+  namespace: openshift-cloud-credential-operator
+spec:
+  cloudTokenPath: /var/run/secrets/openshift/serviceaccount/token
+  providerSpec:
+    apiVersion: cloudcredential.openshift.io/v1
+    kind: AWSProviderSpec
+    statementEntries:
+    - action:
+      - s3:ListBucket
+      - s3:PutObject
+      - s3:GetObject
+      - s3:DeleteObject
+      effect: Allow
+      resource: arn:aws:s3:*:*:*
+    stsIAMRoleARN: arn:aws:iam::${AWS_ACCOUNT_ID}:role/openshift-logging-loki-operator-controller-manager
+  secretRef:
+    name: logging-loki-aws-creds
+    namespace: openshift-logging
+  serviceAccountNames:
+  - logging-loki
+```
+
+#### Azure WIF Support
+
+The Loki Operator will use the provided environment variables (`CLIENT_ID`, `TENANT_ID`, `SUBSCRIPTION_ID`) and create a `CredentialsRequest` resource in the CCO operator namespace per LokiStack resource. The values will be used in the `CredentialsRequest` fields `spec.providerSpec.{AzureClientID,AzureTenantID,AzureSubscriptionID}` respectively. The LokiStack administrator is required to provider an Azure Region via the Azure object storage secret.
```

Comment:
There is a discrepancy between the fields mentioned here and in the [design doc](https://github.com/grafana/loki/pull/11060/files#diff-27b5aa2c87cf76afa33759f0ba55f749b2a54e1d4b1b6c7bc69a2cf1cbfc1c24R69-R76). In the design doc we mention `AzureClientID,AzureTenantID` however CCO seems to care about some other fields `AzureSubscriptionID, AzureRegion`.

Reply:
That's correct and well known because my investigation in the Loki code base did not uncover any usage of the azure subscription id and/or region. I am leaving this open here for the research task in LOG-4545. WDYT?